### ISSUE TITLE
SemanticSegmentationTask MisconfigurationException 

### ISSUE THREAD

**Issue by kethan52** [microsoft/torchgeo#519]
Hi my dataloader does not have validation split while i am running code it's giving the error 
```python
model =SemanticSegmentationTask(
    segmentation_model="fcn",
    encoder_name="resnet18",
    encoder_weights="imagenet",
    in_channels=9,
    num_classes=2,
    num_filters=128,
    loss="ce",
    ignore_zeros=True,
    learning_rate=0.1,
    learning_rate_schedule_patience=0.05,
    
)

trainer = pl.Trainer(
                  gpus=1,
                  callbacks=[checkpoint_callback, early_stopping_callback],
                  logger=[csv_logger],
                  max_epochs=10,
                  precision=32,
                  log_every_n_steps=0.01,
                  max_steps=4,
               )

trainer.fit(model,dl)
```
![Screenshot (1432)](https://user-images.githubusercontent.com/41123619/166162973-b721af07-2fec-4145-97eb-a7c6ba361536.png)

**Comment by adamjstewart** [comment]
I haven't used PyTorch Lightning too much, but I think our trainers may be designed to expect some kind of validation set. Are you able to split your train dataset into train + validation?

**Comment by isaaccorley** [comment]
The Segmentation trainer is currently set to monitor "val_loss" which requires that you have a validation dataloader in your lightning data module. You can either add a validation dataloader or subclass the trainer and override the configure_optimizers method to monitor "train_loss".

**Comment by kethan52** [comment]
Okay thank you soo much !!! can I get the snippet of the code like how to subclass the trainer and override the configure_optimizers method to monitor "train_loss".

**Comment by adamjstewart** [comment]
Try the following:
```python
import torch
from torchgeo.trainers import SemanticSegmentationTask

class TrainSemanticSegmentationTask(SemanticSegmentationTask):
    def configure_optimizers(self):                                                                                                               
        optimizer = torch.optim.Adam(                                                    
            self.model.parameters(), lr=self.hyperparams["learning_rate"]                
        )                                                                                
        return {                                                                         
            "optimizer": optimizer,                                                      
            "lr_scheduler": {                                                            
                "scheduler": ReduceLROnPlateau(                                          
                    optimizer,                                                           
                    patience=self.hyperparams["learning_rate_schedule_patience"],        
                ),                                                                       
                "monitor": "train_loss",                                                   
            },                                                                           
        }       

model = TrainSemanticSegmentationTask(...)
```

**Comment by kethan52** [comment]
Thank you soo much
